pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -209,4 +210,7 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.EnforceNodeAllocatable == nil {
 		obj.EnforceNodeAllocatable = DefaultNodeAllocatableEnforcement
 	}
+	if obj.PreferredProbeIPFamily == "" {
+		obj.PreferredProbeIPFamily = kubeletconfigv1beta1.IPFamilyNone.String()
```

Comment:
I think this setting of the PreferredProbeIPFamily to none is causing CI test failures, since valid options are "ipv4" or "ipv6". We chatted about having this set to whatever the family is of the cluster's service CIDR, but kubelet is started early and it would be very difficult to do this programmatically. Maybe this should default to "ipv4", and we'll need to document that the admin should make sure to start kubelet with a family matching the service CIDR (and remember to modify kubeadm to set this kubelet flag appropriately).

Reply:
What is kubelet expected to do if it finds "none" ?

Reply:
Am I understanding properly that this will intentionally fail validation?  That effectively makes this a new required field, which means old installations might break upon upgrade, which is not allowed.

There has to be a default that is compatible with the old way of doing things.

Reply:
If kubelet finds the PreferredProbeIPFamily set to "none" (meaning no preference on family), it's expected to use the default IP (i.e. the singular PodIP field). This is done on this line: https://github.com/kubernetes/kubernetes/pull/69029/files#diff-0c9236a776261037dc32a136fc6642c1R293
The validation failure isn't intentional. It seems that the validation on this line is missing a check for "none" as a valid value: https://github.com/kubernetes/kubernetes/pull/69029/files#diff-071bb4b0561610e751f5b1137e6caf5fR139

Reply:
@bruceauyeung - Scratch my initial comment. If we keep this knob, the default should be "none" (no preference, i.e. use primary IP family), and the validation should be changed to allow a value of "none".

Reply:
a value of "none" is allowed now. I think the KEP doc should also be updated.

Reply:
@bruceauyeung - I've updated the KEP to reflect this.